### PR TITLE
[2/5] [4] cluster: change order of codes

### PR DESCRIPTION
**Pull request by 0xWOF**
## Changes

- cluster: move ClusterConnection struct declaration to above of impl
- cluster: move set_auto_reconnect

## Effect

- cluster:
    - no public api is changed
    - code is changed but behavior will not be changed

## Note

- this PR is from https://github.com/redis-rs/redis-rs/pull/690
- this PR is depends on https://github.com/redis-rs/redis-rs/pull/707
- you can check actual file changes on https://github.com/0xWOF/redis-rs/pull/16

### CODE REVIEW

**Comment by 0xWOF** [comment]
I rebase it because of PR 3 is merged. After this PR, I'll create PR one by one.

File: redis/src/cluster.rs [hunk 1]
```diff
@@ -94,6 +80,20 @@ impl TlsMode {
     }
 }
 
+/// This is a connection of Redis cluster.
```
Comment (djc) [inline redis/src/cluster.rs:83]: This is an improvement, but I'd prefer to instead move `TlsMode` and its `impl` closer to the bottom of the module.

File: redis/src/cluster.rs [hunk 2]
```diff
@@ -194,6 +187,13 @@ impl ClusterConnection {
         Ok(())
     }
 
+    /// Set an auto reconnect attribute.
```
Comment (djc) [inline redis/src/cluster.rs:190]: I don't see the point? This was already grouped with other `pub fn ..(&self)` methods, so not much point in moving it around within that group?

Comment (0xWOF) [reply redis/src/cluster.rs]: I agree with you. There are no reason to move set_auto_reconnect. I restore it.

File: redis/src/cluster.rs (hunk 1 above)
Comment (djc) [reply redis/src/cluster.rs:65]: Yes, moving `SlotMap` to the bottom makes sense to me as well.

Comment (0xWOF) [reply redis/src/cluster.rs:65]: 👍

File: redis/src/cluster.rs (hunk 2 above)
Comment (0xWOF) [reply redis/src/cluster.rs]: Sure. I drop the commit.

**Review by djc — APPROVED** [review]

**Comment by 0xWOF** [comment]
ci problem is same with previous intermittent one. <https://github.com/redis-rs/redis-rs/pull/704#issuecomment-1302313412>

**Comment by utkarshgupta137** [comment]
IMO, it is better to define everything before it is used in the code, to the extent possible. The purpose of the original commit was to bring the struct & impl of ClusterConnection next to each other. You're of course, free to follow your convention.

File: redis/src/cluster.rs [hunk 3]
```diff
@@ -694,6 +678,22 @@ impl NodeCmd {
...
+    fn from_insecure_flag(insecure: bool) -> TlsMode {
+        if insecure {
+            TlsMode::Insecure
+        } else {
+            TlsMode::Secure
+        }
+    }
+}
+
 impl ConnectionLike for ClusterConnection {
```
[the remark this answers is not in the document]
Comment (0xWOF) [reply redis/src/cluster.rs:697]: Sorry for I misunderstand your purpose. I agree with your way.
@djc how about move `impl ConnectionLike for ClusterConnection` ? if you okay with this way, I want to apply this too.

**Comment by utkarshgupta137** [comment]
Not to mention that the more changes you make early on, the hard it will be to apply future changes. There are a lot of changes in this file, so it would probably be best not to mess with the order of code if it doesn't make a lot of difference.

File: redis/src/cluster.rs (hunk 3 above)
[the remark this answers is not in the document]
Comment (djc) [reply redis/src/cluster.rs:697]: Yes, I agree (I think I already mentioned that?).

**Comment by 0xWOF** [comment]
@utkarshgupta137 I think some changes are can't be avoidable and rebase difficulty is still bearable. So I'll be careful more not to do mis-rebase for now. But your concern is reasonable too. I'll find other better way to satisfy ease-of-rebase and code-change-ability when I can't bearable more.

File: redis/src/cluster.rs (hunk 3 above)
[the remark this answers is not in the document]
Comment (0xWOF) [reply redis/src/cluster.rs]: I apply it. Thanks!

**Review by djc — APPROVED** [review]

**Pull request merged**